### PR TITLE
Support BigDecimal for NUMERIC (Text) data type

### PR DESCRIPTION
**Pull request by EmadAlblueshi**
Fixes https://github.com/vietj/reactive-pg-client/issues/61

@vietj would you please review ?

### CODE REVIEW

File: src/main/java/com/julienviet/pgclient/impl/ArrayTuple.java [hunk 1]
```diff
@@ -91,6 +92,13 @@ public Double getDouble(int pos) {
     return null;
   }
 
+  @Override
+  public BigDecimal getBigDecimal(int pos) {
+    if(get(pos) instanceof BigDecimal)
```
Comment (vietj) [inline src/main/java/com/julienviet/pgclient/impl/ArrayTuple.java]: I think we need also to handle the case where the object is an instanceof Number and convert it

File: src/main/java/com/julienviet/pgclient/impl/RowImpl.java [hunk 2]
```diff
@@ -131,4 +132,9 @@ public OffsetDateTime getOffsetDateTime(String name) {
   public UUID getUUID(String name) {
     return getUUID(desc.columnIndex(name));
   }
+
+  @Override
+  public BigDecimal getBigDecimal(String name) {
+    return getBigDecimal(desc.columnIndex(name));
```
Comment (vietj) [inline src/main/java/com/julienviet/pgclient/impl/RowImpl.java]: we need to handle the case where columnIndex returns -1

**Comment by vietj** [comment]
I think what we are missing is a patch so that NUMERIC type is always a BigInteger and not sometimes to Double

File: src/main/java/com/julienviet/pgclient/impl/RowImpl.java (hunk 2 above)
Comment (EmadAlblueshi) [reply src/main/java/com/julienviet/pgclient/impl/RowImpl.java]: @vietj is this should be handled for `BigDecimal` only ?

**Comment by EmadAlblueshi** [comment]
Now `NUMERIC` will always return `BigDecimal` which handles both cases `Integer/Decimal`

File: src/main/java/com/julienviet/pgclient/impl/RowImpl.java (hunk 2 above)
Comment (vietj) [reply src/main/java/com/julienviet/pgclient/impl/RowImpl.java]: no, it should always be handled, perhaps we should handle it in a separate patch

Comment (EmadAlblueshi) [reply src/main/java/com/julienviet/pgclient/impl/RowImpl.java]: ok

Comment (vietj) [reply src/main/java/com/julienviet/pgclient/impl/RowImpl.java]: it seems actually is should be fine, we should check at least these cases are tested.

Comment (vietj) [reply src/main/java/com/julienviet/pgclient/impl/RowImpl.java]: https://github.com/vietj/reactive-pg-client/issues/68

**Comment by EmadAlblueshi** [comment]
👍

**Pull request merged**